internal/platform/platform.go

```diff
@@ -1,32 +1,82 @@
 package platform
 
 import (
-	"strings"
+	"fmt"
 
 	"github.com/jaypipes/ghw"
+	"github.com/openshift/dpu-operator/internal/daemon/plugin"
 	"sigs.k8s.io/kind/pkg/errors"
 )
 
-func supportedDpuPlatforms() []string {
-	return []string{"IPU Adapter E2100-CCQDA2"}
+type PlatformInfoProvider interface {
 }
 
-type PlatformInfoProvider interface {
-	IsDPU() (bool, error)
+type VendorDetector interface {
+	IsDpuPlatform() (bool, error)
+	VspPlugin(dpuMode bool) *plugin.GrpcPlugin
+	IsDPU(pci ghw.PCIDevice) bool
 }
 
-type PlatformInfo struct{}
+type PlatformInfo struct {
+	Detectors []VendorDetector
+}
 
-func (pi *PlatformInfo) IsDPU() (bool, error) {
-	product, err := ghw.Product()
-	if err != nil {
-		return false, errors.Errorf("Error getting product info: %v", err)
+func NewPlatformInfo() *PlatformInfo {
+	return &PlatformInfo{
+		Detectors: []VendorDetector{
+			NewIntelDetector(),
+			// add more detectors here
+		},
 	}
+}
 
-	for _, model := range supportedDpuPlatforms() {
-		if strings.Contains(product.Name, model) {
+func (pi *PlatformInfo) IsDpu() (bool, error) {
+	for _, detector := range pi.Detectors {
+		isDpu, err := detector.IsDpuPlatform()
+		if err != nil {
+			return false, err
+		}
+		if isDpu {
 			return true, nil
 		}
 	}
 	return false, nil
 }
+
+func (pi *PlatformInfo) listDpuDevices() ([]ghw.PCIDevice, []VendorDetector, error) {
+	pci, err := ghw.PCI()
+	if err != nil {
+		return nil, nil, errors.Errorf("Error getting PCI info: %v", err)
+	}
+
+	if err != nil {
+		return nil, nil, err
+	}
+	var dpuDevices []ghw.PCIDevice
+	var detectors []VendorDetector
+	for _, pci := range pci.ListDevices() {
+		for _, detector := range pi.Detectors {
+			if detector.IsDPU(*pci) {
+				dpuDevices = append(dpuDevices, *pci)
+				detectors = append(detectors, detector)
+				break
+			}
+		}
+	}
+	return dpuDevices, detectors, nil
+}
+
+func (pi *PlatformInfo) VspPlugin(dpuMode bool) (*plugin.GrpcPlugin, error) {
+	if dpuMode {
+		return plugin.NewGrpcPlugin(dpuMode), nil
+	} else {
+		dpuDevices, detectors, err := pi.listDpuDevices()
+		if err != nil {
+			return nil, errors.Errorf("Failed to get VspPlugin from platform: %v", err)
+		}
+		if len(dpuDevices) != 1 {
+			return nil, fmt.Errorf("Failed to detect which vsp to use")
+		}
+		return detectors[0].VspPlugin(dpuMode), nil
```

Comment:
We should make a TODO here. To support more than 1 VSP?

Reply:
Do we want that? I think the error is what you will hit when you want to start adding more DPUs.

Reply:
Some text editors gives a list of TODOs... But I like to document things that appear weird to someone that doesn't have all the history. I.e. someone jumping into the project and not knowing the discussions our team has about what is mandatory and what is TODO in the future.

Reply:
It doesn't have to be called TODO. But just document oddities with the mindset if you knew nothing while reading the code. Then thinking it might be a bug.

Reply:
What if we change the error to: `Multiple DPUs detected. Curreently only supporting 1 DPU per host`

Reply:
Yes that would work too if you add a check for >1.